globus_automate_client/cli/callbacks.py

```diff
@@ -64,6 +65,48 @@ def json_validator_callback(body: str) -> str:
     return body
 
 
+def yaml_validator_callback(body: str) -> str:
+    """
+    A user supplied body can be a properly formatted YAML string or a the name
+    of a file that contains valid YAML.  This validator ensures the user
+    supplies a valid YAML or valid filename containing valid YAML. Returns the
+    valid JSON string.
+    """
+    # Callbacks are run regardless of whether an option was explicitly set.
+    # Handle the scenario where the default value for an option is empty
+    if not body:
+        return body
+
+    # Reading from a file was indicated by prepending the filename with the @
+    # symbol -- for backwards compatability check if the symbol is present and
+    # remove it
+    if body.startswith("@"):
+        body = body[1:]
+
+    body_path = pathlib.Path(body)
+
+    if body_path.exists() and body_path.is_file():
+        with body_path.open() as f:
+            try:
+                yaml_body = yaml.safe_load(f)
+            except yaml.YAMLError as e:
+                raise typer.BadParameter(f"Invalid YAML: {e}")
+    elif body_path.exists() and body_path.is_dir():
+        raise typer.BadParameter("Expected file, received directory")
+    else:
+        try:
+            yaml_body = yaml.safe_load(body)
+        except yaml.YAMLError as e:
+            raise typer.BadParameter(f"Invalid YAML: {e}")
+
+    try:
+        yaml_to_json = json.dumps(yaml_body)
+    except TypeError as e:
+        raise typer.BadParameter(f"Unable to translate to JSON: {e}")
+
+    return yaml_to_json
```

Comment:
This is a good catch - I think we want to standardize all input as it enters the CLI layer and standardizing on json is a good thing.